js/search-tabs-carousel.js

```diff
@@ -0,0 +1,146 @@
+        activeTabButton.innerText = `Explore ${event.target.innerText}`;
+
+        activeTab = newActiveTab;
+    }
+});
+
+const box = document.querySelector('.search_box');
+function containsSearch(course) {
+    if (course['title'].toLowerCase().includes(box.value.toLowerCase())
+        || course['headline'].toLowerCase().includes(box.value.toLowerCase()))
+        return true;
+
+    for (let instructor of course['instructors']) {
+        if (instructor['name'].toLowerCase().includes(box.value.toLowerCase()))
+            return true;
+    }
+    return false;
+}
+
+
+
+function getNumberOfCardsPerSlide(width) {
+    if (width >= 1450)
+        return 5;
+    else if (width >= 1200)
+        return 4;
+    else if (width >= 992)
+        return 3;
+    else if (width >= 666)
+        return 2;
+    else
+        return 1;
+}
+
+
+const indicators = document.querySelector('.carousel-indicators');
+const slides = document.querySelector('.carousel-inner');
+
+let cardsPerSlide = 0;
+function buildCourses(tab) {
+    indicators.innerHTML = '';
+    slides.innerHTML = '';
+
+    cardsPerSlide = getNumberOfCardsPerSlide(window.innerWidth);
+
+    let temp = courses[tab]['courses'].filter(containsSearch);
+    for (let i = 0; i < temp.length; i += cardsPerSlide) {
+        const indicator = document.createElement('button');
+        indicator.setAttribute('type', 'button');
+        indicator.setAttribute('data-bs-target', '#carouselExampleIndicators');
+        indicator.setAttribute('data-bs-slide-to', `${i / cardsPerSlide}`);
+        indicator.setAttribute('aria-label', `Slide ${(i / cardsPerSlide) + 1}`);
+
+        if (i == 0) {
+            indicator.setAttribute('class', 'active');
+            indicator.setAttribute('aria-current', 'true');
+        }
+
+        const carouselItem = document.createElement('div');
+
+        carouselItem.classList.add('carousel-item');
+        if (i === 0) {
+            carouselItem.classList.add('active');
+        }
+
+        const individualSlide = document.createElement('div');
+        individualSlide.classList.add('courses-inject', 'row', 'd-flex', 'flex-nowrap', 'justify-content-start', 'm-auto');
+
+        for (let k = i; k < Math.min(i + cardsPerSlide, temp.length); k++) {
+            let course = temp[k];
+            individualSlide.innerHTML += `
+                <div class="mx-1">
+                    <img src="${course['image']}" alt="course ${course['id']}">
+                    <h3 class="title">${course['title']}</h3>
+                    <h4>${course['instructors'][0]['name']}</h4>
+                    <h3 class="price">$${course['price']}</h3>
+                </div>
+
+            `;
+        }
+        carouselItem.append(individualSlide);
+        slides.append(carouselItem);
+
+        indicators.append(indicator);
+    }
+}
+
+addEventListener('resize', (event) => {
+    if (cardsPerSlide != getNumberOfCardsPerSlide(window.innerWidth)) {
```

Comment:
you can use the [debounced](https://www.geeksforgeeks.org/debouncing-in-javascript/) function to improve performance you can [lodash library](https://www.npmjs.com/package/lodash.debounce).